### PR TITLE
scheduling: fix duplicate checks for number of enabled queue sort plugin

### PR DESCRIPTION
**Pull request by NoicFank**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
remove duplicate checks for number of enabled queue sort plugin, only exactly one queue sort plugin is approved.

#### Which issue(s) this PR fixes:

NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Comment by NoicFank** [comment]
/sig/scheduling

**Review by sanposhiho — APPROVED** [review]
nice catch.
/lgtm

**Comment by NoicFank** [comment]
@sanposhiho PTAL, thanks.

**Review by sanposhiho — COMMENTED** [review]
/lgtm

File: pkg/scheduler/framework/runtime/framework.go [hunk 1]
```diff
@@ -330,10 +330,6 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 		}
 	}
 
-	if len(f.queueSortPlugins) != 1 {
```
Comment (kerthcet) [inline pkg/scheduler/framework/runtime/framework.go:333]: It seems we will not modify `queueSortPlugins` and `bindPlugins` below, so I prefer to keep this here, then we can quick return when meeting errors.

Comment (damemi) [reply pkg/scheduler/framework/runtime/framework.go:333]: +1 this is here for quick failures

File: pkg/scheduler/framework/runtime/framework.go [hunk 2]
```diff
@@ -347,13 +343,13 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 	}
 
 	if len(f.queueSortPlugins) == 0 {
-		return nil, fmt.Errorf("no queue sort plugin is enabled")
+		return nil, fmt.Errorf("no queue sort plugin is enabled for profile with scheduler name %q", profile.SchedulerName)
```
Comment (damemi) [inline pkg/scheduler/framework/runtime/framework.go:346]: why remove the scheduler name from all of these checks?

**Review by damemi — COMMENTED** [review]
just for reference, this check was intentionally moved from Profile validation in https://github.com/kubernetes/kubernetes/pull/105611/files#diff-e0fad3cb95eba326e410d482334119367325838fc0f1434825a3e481fc055f2aL99-L100, since that validation happens earlier it was put ahead of other failures to catch the same fail-fast use case and be consistent with the existing error propagation (which was caught by CI)

Looks like CI passes without it, so I'm not sure if other code has changed around it but I still think having a basic check like this early is good

File: pkg/scheduler/framework/runtime/framework.go (hunk 1 above)
Comment (NoicFank) [reply pkg/scheduler/framework/runtime/framework.go:333]: yes, thanks for reviews.

File: pkg/scheduler/framework/runtime/framework.go (hunk 2 above)
Comment (NoicFank) [reply pkg/scheduler/framework/runtime/framework.go]: ah, scheduler name is added here.

File: pkg/scheduler/framework/runtime/framework.go (hunk 1 above)
Comment (kerthcet) [reply pkg/scheduler/framework/runtime/framework.go:333]: Maybe simplify as 
```
if len(f.queueSortPlugins) != 1 {
    return nil, fmt.Errorf("only one queue sort plugin required for profile with scheduler name %q, but got %d", profile.SchedulerName, len(f.queueSortPlugins))
}
```

Comment (NoicFank) [reply pkg/scheduler/framework/runtime/framework.go:333]: ok, for sure. thanks quick review.

**Review by sanposhiho — APPROVED** [review]
/lgtm

**Comment by NoicFank** [comment]
@Huang-Wei please review the code, thanks.

**Comment by Huang-Wei** [comment]
/approve

Thanks @NoicFank .

**Comment by NoicFank** [comment]
- 加微信，约时间。- 使用腾讯视频面交作业，链接微信给你。

**Pull request merged**